react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { imageIndex: 0 };
+    this.clickedRight = this.clickedRight.bind(this);
+    this.clickedLeft = this.clickedLeft.bind(this);
+    this.clickedDot = this.clickedDot.bind(this);
+  }
+
+  clickedRight() {
+    if ((this.props.images.length - 1) === parseInt(this.state.imageIndex)) {
+      this.setState({ imageIndex: 0 });
+    } else {
+      this.setState({ imageIndex: parseInt(this.state.imageIndex + 1) });
+    }
+    this.timeReset();
+    this.componentDidMount();
```

Comment:
Try to refactor your code so that you do not call `componentDidMount`, that method should only be used by React.